tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -557,6 +557,11 @@ auto check_exact_per_strategy(IntS tap_pos_any, IntS tap_pos_min, IntS tap_pos_m
         }
     };
 }
+auto normalized_lerp(IntS value, IntS start, IntS stop) {
+    REQUIRE(start != stop);
+    return (static_cast<double>(value) - static_cast<double>(start)) /
+           (static_cast<double>(stop) - static_cast<double>(start));
```

Comment:
`stop-start` sounds like cacheable ;)
Also would any check be necessary? `stop > start` is assumed? `value` falls in range assumed?

Reply:
if `stop < start`, the logic remains valid due to cancelling minus signs. they just cannot be equal to eachother

Reply:
> if `stop < start`, the logic remains valid due to cancelling minus signs. they just cannot be equal to eachother.

Mathematically yes, but semantically? I was more asking about whether both could well be the case.

> if `value \nin [start, stop]` it's `linear extrapolation` instead of `linear interpolation` which is still the same logic

Interesting that this function does more that what it should ;)

Reply:
> ... whether both could well be the case.
[should be read]
Whether `start < stop` and `start> stop` could be the case in reality and that they both mean valid configurations.

Reply:
> > ... whether both could well be the case.
> 
> [should be read] Whether `start < stop` and `start> stop` could both be the case in reality and that they both mean valid configurations.

yes, definitely. `tap_min < tap_max` and `tap_min > tap_max` are definitely both valid configurations.

> > if `value \nin [start, stop]` it's `linear extrapolation` instead of `linear interpolation` which is still the same logic
> 
> Interesting that this function does more that what it should ;)

well, `std::lerp` is a standard function for a reason 😬 the problem is that it doesn't serve our needs in this particular case because we need it normalized here (otherwise, the logic becomes explicitly dependent on `tap_min` and `tap_max` which is what we want to avoid in the first place)